<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1112</width>
    <height>748</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Line Rendering with Depth-Dependent Halos</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QWidget" name="controls" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="2" column="1">
        <widget class="QProgressBar" name="progressBar">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="maximum">
          <number>100</number>
         </property>
         <property name="value">
          <number>0</number>
         </property>
         <property name="textVisible">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="labelTop">
         <property name="text">
          <string>No data loaded</string>
         </property>
        </widget>
       </item>
       <item row="2" column="3">
        <widget class="QPushButton" name="generateTestDataButton">
         <property name="maximumSize">
          <size>
           <width>160</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="text">
          <string>Generate Test Data</string>
         </property>
        </widget>
       </item>
       <item row="2" column="2">
        <widget class="QSpinBox" name="testDataNumVerticesSpinBox">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="toolTip">
          <string>number of vertices to generate for test data</string>
         </property>
         <property name="inputMethodHints">
          <set>Qt::ImhDigitsOnly</set>
         </property>
         <property name="suffix">
          <string> vertices</string>
         </property>
         <property name="prefix">
          <string/>
         </property>
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>999999999</number>
         </property>
         <property name="value">
          <number>100000</number>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="2" column="0">
     <layout class="QGridLayout" name="glLayout"/>
    </item>
    <item row="1" column="0">
     <widget class="Line" name="line">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </item>
    <item row="2" column="1">
     <widget class="QWidget" name="widget_2" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>200</width>
        <height>0</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>200</width>
        <height>16777215</height>
       </size>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QGroupBox" name="groupBox_4">
         <property name="title">
          <string>Drawing</string>
         </property>
         <widget class="QComboBox" name="comboBoxDrawMode">
          <property name="geometry">
           <rect>
            <x>95</x>
            <y>40</y>
            <width>69</width>
            <height>22</height>
           </rect>
          </property>
          <item>
           <property name="text">
            <string>Lines</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Points</string>
           </property>
          </item>
         </widget>
         <widget class="QLabel" name="labelDrawMode">
          <property name="geometry">
           <rect>
            <x>15</x>
            <y>42</y>
            <width>60</width>
            <height>20</height>
           </rect>
          </property>
          <property name="text">
           <string>Mode</string>
          </property>
         </widget>
         <widget class="QDoubleSpinBox" name="spinBoxLineTriangleStripWidth">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="geometry">
           <rect>
            <x>95</x>
            <y>69</y>
            <width>71</width>
            <height>22</height>
           </rect>
          </property>
          <property name="toolTip">
           <string>total line width (black line stroke + white halo)</string>
          </property>
          <property name="decimals">
           <number>4</number>
          </property>
          <property name="minimum">
           <double>0.000000000000000</double>
          </property>
          <property name="maximum">
           <double>10.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.005000000000000</double>
          </property>
          <property name="value">
           <double>0.030000000000000</double>
          </property>
         </widget>
         <widget class="QLabel" name="labelLineTriangleStripWidth">
          <property name="geometry">
           <rect>
            <x>15</x>
            <y>71</y>
            <width>60</width>
            <height>20</height>
           </rect>
          </property>
          <property name="text">
           <string>Width</string>
          </property>
         </widget>
         <widget class="QDoubleSpinBox" name="spinBoxLineWidthDepthCueingFactor">
          <property name="geometry">
           <rect>
            <x>95</x>
            <y>125</y>
            <width>71</width>
            <height>22</height>
           </rect>
          </property>
          <property name="toolTip">
           <string>draw black lines thinner with increasing depth</string>
          </property>
          <property name="maximum">
           <double>10.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.050000000000000</double>
          </property>
          <property name="value">
           <double>1.000000000000000</double>
          </property>
         </widget>
         <widget class="QLabel" name="labelLineWidthPercentageBlack">
          <property name="geometry">
           <rect>
            <x>15</x>
            <y>99</y>
            <width>71</width>
            <height>20</height>
           </rect>
          </property>
          <property name="text">
           <string>Stroke</string>
          </property>
         </widget>
         <widget class="QDoubleSpinBox" name="spinBoxLineHaloMaxDepth">
          <property name="geometry">
           <rect>
            <x>95</x>
            <y>153</y>
            <width>71</width>
            <height>22</height>
           </rect>
          </property>
          <property name="toolTip">
           <string>maximum depth displacement for white halo. higher means lines close in depth are occluded by less halo and appear more bundled.</string>
          </property>
          <property name="decimals">
           <number>4</number>
          </property>
          <property name="minimum">
           <double>0.000000000000000</double>
          </property>
          <property name="maximum">
           <double>1.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.005000000000000</double>
          </property>
          <property name="value">
           <double>0.020000000000000</double>
          </property>
         </widget>
         <widget class="QLabel" name="labelLineHaloMaxDepth">
          <property name="geometry">
           <rect>
            <x>15</x>
            <y>155</y>
            <width>81</width>
            <height>20</height>
           </rect>
          </property>
          <property name="text">
           <string>Bundling</string>
          </property>
         </widget>
         <widget class="QLabel" name="labelLineWidthDepthCueingFactor">
          <property name="geometry">
           <rect>
            <x>15</x>
            <y>127</y>
            <width>91</width>
            <height>20</height>
           </rect>
          </property>
          <property name="text">
           <string>Thinning</string>
          </property>
         </widget>
         <widget class="QDoubleSpinBox" name="spinBoxLineWidthPercentageBlack">
          <property name="geometry">
           <rect>
            <x>95</x>
            <y>97</y>
            <width>71</width>
            <height>22</height>
           </rect>
          </property>
          <property name="toolTip">
           <string>percentage of line drawn as black stroke (rest is white halo)</string>
          </property>
          <property name="decimals">
           <number>2</number>
          </property>
          <property name="maximum">
           <double>1.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.050000000000000</double>
          </property>
          <property name="value">
           <double>0.300000000000000</double>
          </property>
         </widget>
         <widget class="QPushButton" name="pushButtonRestoreDefaults">
          <property name="geometry">
           <rect>
            <x>95</x>
            <y>181</y>
            <width>71</width>
            <height>22</height>
           </rect>
          </property>
          <property name="text">
           <string>Reset</string>
          </property>
         </widget>
         <widget class="QCheckBox" name="checkBoxEnableClipping">
          <property name="geometry">
           <rect>
            <x>15</x>
            <y>224</y>
            <width>130</width>
            <height>26</height>
           </rect>
          </property>
          <property name="text">
           <string>Enable Clipping</string>
          </property>
         </widget>
         <widget class="QSlider" name="horizontalSliderClipPlaneDistance">
          <property name="geometry">
           <rect>
            <x>15</x>
            <y>277</y>
            <width>151</width>
            <height>30</height>
           </rect>
          </property>
          <property name="minimum">
           <number>0</number>
          </property>
          <property name="maximum">
           <number>100</number>
          </property>
          <property name="value">
           <number>50</number>
          </property>
          <property name="tracking">
           <bool>true</bool>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="invertedAppearance">
           <bool>false</bool>
          </property>
          <property name="invertedControls">
           <bool>false</bool>
          </property>
          <property name="tickPosition">
           <enum>QSlider::NoTicks</enum>
          </property>
          <property name="tickInterval">
           <number>0</number>
          </property>
         </widget>
         <widget class="QPushButton" name="pushButtonSetClipPlaneNormal">
          <property name="geometry">
           <rect>
            <x>15</x>
            <y>255</y>
            <width>151</width>
            <height>22</height>
           </rect>
          </property>
          <property name="text">
           <string>Update Clip Direction</string>
          </property>
         </widget>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox_5">
         <property name="title">
          <string>GPU</string>
         </property>
         <widget class="QLabel" name="label_9">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>70</y>
            <width>91</width>
            <height>16</height>
           </rect>
          </property>
          <property name="text">
           <string>Used Memory (MB)</string>
          </property>
         </widget>
         <widget class="QLCDNumber" name="usedMemLCD">
          <property name="geometry">
           <rect>
            <x>110</x>
            <y>67</y>
            <width>64</width>
            <height>23</height>
           </rect>
          </property>
         </widget>
         <widget class="QLabel" name="label_10">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>20</y>
            <width>51</width>
            <height>21</height>
           </rect>
          </property>
          <property name="text">
           <string/>
          </property>
         </widget>
         <widget class="QLabel" name="label_11">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>38</y>
            <width>91</width>
            <height>21</height>
           </rect>
          </property>
          <property name="text">
           <string>Total Memory (MB)</string>
          </property>
         </widget>
         <widget class="QLCDNumber" name="memSizeLCD">
          <property name="geometry">
           <rect>
            <x>110</x>
            <y>37</y>
            <width>64</width>
            <height>23</height>
           </rect>
          </property>
          <property name="segmentStyle">
           <enum>QLCDNumber::Filled</enum>
          </property>
         </widget>
         <widget class="QLCDNumber" name="fpsLCD">
          <property name="geometry">
           <rect>
            <x>110</x>
            <y>97</y>
            <width>64</width>
            <height>23</height>
           </rect>
          </property>
         </widget>
         <widget class="QLabel" name="label_12">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>102</y>
            <width>91</width>
            <height>16</height>
           </rect>
          </property>
          <property name="text">
           <string>FPS</string>
          </property>
         </widget>
         <widget class="QLabel" name="labelGraphicsDeviceInfo">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>133</y>
            <width>171</width>
            <height>171</height>
           </rect>
          </property>
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="font">
           <font>
            <pointsize>9</pointsize>
           </font>
          </property>
          <property name="text">
           <string>Graphics Device: None</string>
          </property>
          <property name="scaledContents">
           <bool>false</bool>
          </property>
          <property name="alignment">
           <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
          </property>
          <property name="wordWrap">
           <bool>true</bool>
          </property>
         </widget>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1112</width>
     <height>25</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen"/>
    <addaction name="separator"/>
    <addaction name="actionClose"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <action name="actionOpen">
   <property name="text">
    <string>Open ...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionClose">
   <property name="text">
    <string>Close</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+W</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
